api/types/authentication.go

```diff
@@ -242,29 +231,32 @@ func (c *AuthPreferenceV2) GetRequireSessionMFA() bool {
 	return c.Spec.RequireSessionMFA
 }
 
+// setStaticFields sets basic defaults
+func (c *AuthPreferenceV2) setStaticFields() {
+	c.Kind = KindClusterAuthPreference
+	c.Version = V2
+}
+
 // CheckAndSetDefaults verifies the constraints for AuthPreference.
 func (c *AuthPreferenceV2) CheckAndSetDefaults() error {
-	// make sure we have defaults for all metadata fields
-	err := c.Metadata.CheckAndSetDefaults()
-	if err != nil {
-		return trace.Wrap(err)
-	}
-	if c.Version == "" {
-		c.Version = V2
-	}
+	c.setStaticFields()
 
-	// Make sure origin value is always set.
-	if c.Origin() == "" {
-		c.SetOrigin(OriginDynamic)
+	if c.Metadata.Name == "" {
+		c.Metadata.Name = MetaNameClusterAuthPreference
```

Comment:
Shouldn't these name values for singleton resources be considered as "static fields" as well?

Reply:
Sure, I'll update it to be `ResourceHeader` and `Metadata` fields.